### PR TITLE
adding logic to auto aprv submission without cmnt

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
*Issue #:*
https://github.com/bcgov/met-public/issues/1370

*Description of changes:*
changes to auto approve submissions which:
- do not have a textarea or textfield in the submission json
- has textarea or textfield in the submission json but have a null value

changes to exclude auto approved submissions from comment listing page as they do not have a comment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

**Review by jadmsaadaot — APPROVED** [review]

**Review by djnunez-aot — APPROVED** [review]

File: met-api/src/met_api/models/submission.py [hunk 1]
```diff
@@ -53,6 +53,16 @@ def get_by_survey_id(cls, survey_id) -> List[SubmissionSchema]:
     @classmethod
     def create(cls, submission: SubmissionSchema, session=None) -> Submission:
         """Save submission."""
+        has_comments = cls.__check_if_submission_has_comments(submission)
```
Comment (saravanpa-aot) [inline met-api/src/met_api/models/submission.py:56]: can you add some comments why we are setting to approved if no comments?
just to explain the logic..

**Review by saravanpa-aot — COMMENTED** [review]
Can you add a test for this?
Do a sumbission ,
Assert the status is approved

File: met-api/src/met_api/models/submission.py [hunk 2]
```diff
@@ -72,6 +84,31 @@ def create(cls, submission: SubmissionSchema, session=None) -> Submission:
             session.flush()
         return new_submission
 
+    @staticmethod
```
Comment (saravanpa-aot) [inline met-api/src/met_api/models/submission.py:87]:
```
@staticmethod
def __check_if_submission_has_comments(submission: SubmissionSchema):
    """Check if comment exists."""
    submission_json = submission.get('submission_json', {})
    text_fields = ['simpletextarea', 'simpletextfield']

    for field in text_fields:
        if field in submission_json and len(submission_json[field]) > 0:
            return True

    return False
```

Comment (saravanpa-aot) [reply met-api/src/met_api/models/submission.py:87]: can we simply this?
do check the logic in detail please..i am not sure 100% if it works

File: met-api/src/met_api/models/submission.py (hunk 1 above)
Comment (VineetBala-AOT) [reply met-api/src/met_api/models/submission.py:61]: added the comments, thanks

File: met-api/src/met_api/models/submission.py (hunk 2 above)
Comment (VineetBala-AOT) [reply met-api/src/met_api/models/submission.py:92]: Changed the logic as suggested, thanks.

**Review by saravanpa-aot — APPROVED** [review]
great job!!

**Pull request merged**